[TPH777] iP

TPH777 opened the pull request.

TPH777 committed:
Change chatbot name to Iris

TPH777 committed:
Echo user command

TPH777 committed:
Store text and return when requested

Store user texts in a String array of fixed size of 100.
When user command "list", print the texts.

TPH777 committed:
Store user text and list out when commanded

Store text in a String array of a fixed size of 100.
When user commands "list", iterate through the array to print out the
stored texts.

TPH777 committed:
Add mark and unmark functionality

Added marked/unmarked boxes beside list
Abstraction of codes within chat to individual functions

TPH777 committed:
Add mark and unmark tasks functionality

Added checkbox beside list to show whether is it marked
Abstraction of code in main to individual functions

TPH777 committed:
Wrap line to keep length shorter than 120chars

TPH777 committed:
Refactor and optimise code

Remove happy path nesting
Wrap lines for print statements
Removes complex expressions

TPH777 committed:
Add task type and date

Added todo, deadline and event command on top of list, mark and bye,
otherwise it will print invalid command

TPH777 committed:
Change enum to individual classes for task type

Separate date from description
Overriden toString method to have a more meanngful string representation
of task

TPH777 committed:
Add I/O redirection testing

Added input and expected testcases
Modified file path for runtest.bat

TPH777 committed:
Optimise and refactor code

Used switch statements and String.split method for clearer flow
Separated mark and unmark functions

CheeKiangg commented on src/main/java/Iris.java:
```diff
@@ -0,0 +1,143 @@
+            }
+            tasks[numOfTasks++] = newTask;
+            printAddTaskMessage(newTask);
+        } catch (RuntimeException e) { // For missing description
```
I like that you added a comment to improve the readability of what exception was being thrown.

CheeKiangg replied:
perhaps you could try abstracting addTask function into 2 smaller functions to improve readability.

CheeKiangg commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,51 @@
+			this.date = "("
+					+ formatDate(textParts[1])
+					+ formatDate(textParts[2])
+					+ ")";
```
you might want to add  "//fallthrough" comment at the end of case 3

CheeKiangg commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,51 @@
+					+ formatDate(textParts[1])
+					+ formatDate(textParts[2])
+					+ ")";
+		}
```
perhaps you could also add in a default statement in the event that all switch cases is false.

CheeKiangg reviewed (commented):
Overall, good job so far. 
Some minor changes requested. Some of the classes are getting a bit long. Consider splitting them into smaller classes if possible.

YukeeHong commented on src/main/java/Iris.java:
```diff
@@ -0,0 +1,143 @@
+        }
+    }
+
+    private static Task getTaskToEdit(String text) {
```
Good use of exception handling to catch missing and invalid task numbers. The code makes it clear when the user provides incomplete information.

YukeeHong commented on src/main/java/Iris.java:
```diff
@@ -0,0 +1,143 @@
+                + " tasks in the list");
+    }
+
+    public static void markTask(String text) {
```
To avoid repetition in markTask and unmarkTask, consider creating a helper function that handles both marking and unmarking tasks.

YukeeHong commented on src/main/java/Iris.java:
```diff
@@ -0,0 +1,143 @@
+import java.util.Scanner;
+
+public class Iris {
+    public static Task[] tasks = new Task[100];
```
For better scalability and flexibility, consider using ArrayList<Task> instead of a fixed-size array. This will eliminate the need to check if the array is full and allows dynamic resizing when adding tasks.

YukeeHong commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,51 @@
+public class Task {
```
The task creation logic could be simplified further by encapsulating task type creation into separate factory methods, reducing the size of the addTask method and improving code readability.

TPH777 committed:
Accept more than two date parts

Removed the limit for 3 text parts (or 2 date parts)
Used loops instead of switch to deal with the change more efficiently

TPH777 committed:
Modify code after reviewing peer reviews

Removed redundant try catch in addTask
Merged mark and unmark task to one function in both Task and Iris

TPH777 committed:
Edit error logic and output

Moved invalid command check from addTask to default in chat
Replace all error message with one that is more specific and has more
characteristics

TPH777 committed:
Add test cases and modify expected output

TPH777 committed:
Add exceptions class specific to Iris

TPH777 committed:
Merge branch 'branch-Level-5'

TPH777 committed:
Organise into packages

Split into exceptions, method and task packages for better readability

TPH777 committed:
Merge branch 'branch-A-Packages'

TPH777 committed:
Add delete command

Share the same function, getTaskNum, as mark to have error handling.
Already used dynamic array, ArrayList.

TPH777 committed:
Add save and continue functionality

When user exit (bye), the tasks will be serialised into the
iris.txt file.
When user start the program, the iris.txt file will be deserialised back to an ArrayList.

TPH777 committed:
Merge branch 'branch-Level-6'

TPH777 committed:
Merge branch 'branch-Level-7'

tzerbin commented on src/main/java/Iris.java:
```diff
@@ -0,0 +1,56 @@
+        System.out.println("---------------------------------------------");
+    }
+
+    public static boolean chat(String text) {
```
It would be good to leave comments on why this method is returning a boolean.
We will discuss comments in Week 7.

tzerbin replied:
Perhaps name the method with a better verb?

tzerbin commented on src/main/java/Iris.java:
```diff
@@ -0,0 +1,56 @@
+        case "":
+            System.out.println("Tell me your needs! I'm here to help!");
+            break;
+        case "bye":
```
Would it be possible to remove the need of magic strings and numbers in your entire code base?

tzerbin commented on src/main/java/exceptions/IrisExceptions.java:
```diff
@@ -0,0 +1,7 @@
+package exceptions;
+
+public class IrisExceptions extends Exception {
```
Is there a reason why the class is named in the plural form?

tzerbin reviewed (commented):
Good attempt in following the guidelines!
You would want to remove magic strings and numbers in your codebase, together with changing some nits on method and variable naming.

TPH777 committed:
Improve code readability

Follow coding standard (IrisExceptions and chat)
Remove magic strings

TPH777 committed:
Refactor code to classes

Have abstract class Command that is used by different commands and get
executed differently.
Have Ui class that take cares of user input and terminal output
Have Storage class to handle the loading and saving of the task list
Have TaskList class to represent ArrayList<Task>
Have Parser class to filter the commands

TPH777 committed:
Add find command

Used streams to filter and print out the task that the user want to find

TPH777 committed:
Remove erroneous character

TPH777 committed:
Add JavaDoc comments

TPH777 committed:
Add date command and change date format in task

Accept only one date input but still allow for different prepositions
like on, by, from
DateCommand when executed will print the task before and after its set
date

TPH777 committed:
Merge pull request #1 from TPH777/branch-Level-8

Add date command and change date format in task

TPH777 committed:
Merge branch 'master' into branch-Level-9

TPH777 committed:
Merge pull request #2 from TPH777/branch-Level-9

Branch level 9

TPH777 committed:
Merge branch 'master' into branch-A-JavaDoc

TPH777 committed:
Delete iris.txt

TPH777 committed:
Remove iris.txt from git

TPH777 committed:
Merge branch 'master' of https://github.com/TPH777/ip

TPH777 committed:
Modify ReadMe

TPH777 committed:
Fix bug

TPH777 committed:
Add user guide for Iris

TPH777 committed:
Add more javaDoc comments

TPH777 committed:
Delete iris.txt

TPH777 committed:
Improve ui

TPH777 committed:
Merge branch 'master' of https://github.com/TPH777/ip

TPH777 committed:
Refactor messages to Ui

Move exception messages to a class
Create function for each printing statement in Ui

TPH777 committed:
Add javadoc comments

TPH777 committed:
Move printDeleteTaskMessage to Ui

TPH777 committed:
Fix minor bugs

Include empty description exception
Trim commands empty spaces for better useabilty
